### PR TITLE
SONARJS-837 Rule: Functions should use "return" consistently

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: javascript-checks/src/main/java/org/sonar/javascript/checks/ConsistentReturnsCheck.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      BlockTree body = (BlockTree) functionTree.body();
+      FunctionReturns functionReturns = getFunctionReturns(body);
+
+      if (functionReturns.containsReturnWithoutValue && functionReturns.containsReturnWithValue) {
+        raiseIssue(tree, functionReturns, body);
+      }
+    }
+  }
+
+  private void raiseIssue(Tree tree, FunctionReturns functionReturns, BlockTree body) {
```
Comment (pynicolas) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/ConsistentReturnsCheck.java]: @vilchik-elena I suggest to rename `tree` to `functionTree`.

File: javascript-checks/src/main/java/org/sonar/javascript/checks/ConsistentReturnsCheck.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    return functionReturns;
+  }
+
+  private static boolean isThrowStatement(Tree lastElement) {
+    return ((JavaScriptTree) lastElement).getParent().is(Kind.THROW_STATEMENT);
+  }
+
+  private static boolean isUnreachableBlock(CfgBlock cfgBlock, ControlFlowGraph cfg) {
```
Comment (pynicolas) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/ConsistentReturnsCheck.java]: @vilchik-elena This method seems to return the opposite of what its name suggests.

Comment (vilchik-elena) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/ConsistentReturnsCheck.java]: @pynicolas thanks!

File: javascript-checks/src/main/java/org/sonar/javascript/checks/ConsistentReturnsCheck.java (hunk 1 above)
Comment (vilchik-elena) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/ConsistentReturnsCheck.java]: @pynicolas ok

**Pull request merged**